internal/mturk/step.go

```diff
@@ -545,16 +545,16 @@ func (s *Session) endMTurkHITStep(ctx context.Context, project *ent.Project, run
 			if !found {
 				p, err = p.Update().
 					AddProjects(project).
-					AddSteps(stepRun).
 					Save(ctx)
 				if err != nil {
-					log.Error().Msgf("could not update participant with workerID %s for stepRun %s", *assignment.WorkerId, stepRun.ID)
+					log.Error().Msgf("could not update participant project with workerID %s for stepRun %s", *assignment.WorkerId, stepRun.ID)
 				}
 			}
 
 			if p.Uninitialized != nil && *p.Uninitialized != false {
 				_, err = p.Update().
 					SetUninitialized(false).
+					AddSteps(stepRun).
```

Comment:
We only add the stepRun if they were uninitialized? I think the steprun should always be added if it's missing, initialized or not.